compiler/rustc_target/src/abi/mod.rs

```diff
@@ -1334,7 +1394,7 @@ impl<'a, Ty> TyAndLayout<'a, Ty> {
         C: HasDataLayout,
     {
         match self.abi {
-            Abi::Scalar(scalar) => scalar.value.is_float(),
+            Abi::Scalar(scalar) => matches!(scalar.primitive(), F32 | F64),
```

Comment:
Nit: might as well add `is_float_primitive` to the `Scalar`. Seen this pattern a couple times already.